### DISCUSSION TITLE
WebdriverIO can't find any elements on new page after click

### DISCUSSION THREAD

**Discussion by RevNight** [webdriverio/webdriverio#6735] [Q&A] [answered]
UPDATE: On the new page, I did a console.log(browser.getUrl()) and it showed the URL of URL 1. I'm on URL 2! What I see in front of me is not what the crawler is seeing. Is this a form of bot protection?

Original:
So I am having a frustrating error in WebdriverIO (Visual Studio Code, node.js, javascript, et al). I have a script that starts on one website and through a series of IF/THENs, clicks on a button to lead it to a new website. This works just fine. But when I am on the new website, none of my code works because none of my selectors can find the elements.

But here is the kicker: When I create another script that goes to this page directly, the exact same code works. All elements are able to be found, interacted with, getText/clicked everything.

Why does the exact same code work only when the page is opened directly, and not when it's opened from a previous pages' button click.

    console.log('INTERMEDIARY PAGE')
    browser.pause(2000)
    let Title = $("(//h1[contains(@class,'cs-entry__title')])")
    console.log(Title)
    let ProductButton = $("(//*[contains(@class,'button')])[1]")
    ProductButton.click()
"Error: Can't call click on element with selector "(//*[contains(@class,'button')])[1]" because element wasn't found"

**Comment by RevNight** [comment]
Ok I still don't know what's causing this, but I did find a workaround. Instead of clicking on the link (DIV tag), do a getAttribute('href') of it instead. Then open it on your own. 

```
<a href="LINK" rel="noopener noreferrer" target="_blank" role="link" class="css-4rbku5 css-18t94o4 css-1dbjc4n r-1loqt21 r-18u37iz r-16y2uox r-1wtj0ep r-1ny4l3l r-o7ynqc r-6416eg">
```
```
    let button = $("(//a[contains(@class,'css-4rbku5 css-18t94o4 css-1dbjc4n r-1loqt21 r-18u37iz r-16y2uox r-1wtj0ep r-1ny4l3l r-o7ynqc r-6416eg')])[2]")
    let attribute = button.getAttribute('href')
    console.log(attribute) --> prints out LINK
    browser.url(attribute)
```

**Comment by christian-bromann** [comment, marked as answer]
Is `(//*[contains(@Class,'button')])[1]` a valid xpath selector?

**Reply by RevNight** [reply]
Yep! I always test this beforehand. Inspect -> Ctrl - F --> paste or write your selector in there. If you see the yellow highlight of code, it's good.
